README.md

```diff
@@ -1,151 +1,177 @@
+    senderAddress,
+    receiver,
+    partnerAddress,
+    partnerFeeBps,
+  }
+);
 
-const allowance = await paraSwap.getAllowance(userAddress, tokenAddress);
+const transactionResponse = await signer.sendTransaction(txParams);
+const transactionReceipt = await transactionResponse.wait();
 ```
 
-##### To approve an ERC20
+## Playground
+Interact with the ParaSwap SDK in CodeSandbox playground [here](https://codesandbox.io/s/gallant-flower-7yuker?file=/src/App.tsx)
 
-```javascript
-const paraSwap = new ParaSwap().setWeb3Provider(web3Provider);
+## Bundle Optimization
+For bundle-size savvy developers, you can construct a lightweight version of this sdk to bring only the functions you need.
 
-const txHash = await paraSwap.approveToken(amount, userAddress, tokenAddress);
-```
+For e.g. for fetching rate and allowance only
 
-##### To build and sign a transaction
+```typescript
+import { constructPartialSDK, constructFetchFetcher, constructGetRate, constructGetAllowance } from '@paraswap/sdk';
 
-```javascript
-const srcToken = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
-const destToken = '0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359';
-const srcAmount = '1000000000000000000'; //The source amount multiplied by its decimals
-const senderAddress = '0xfceA770875E7e6f25E33CEa5188d12Ef234606b4';
-const receiver = '0x8B4e846c90a2521F0D2733EaCb56760209EAd51A'; // Useful in case of swap and transfer
-const referrer = 'my-company-or-nick-name';
+const fetcher = constructFetchFetcher(window.fetch)
 
-const txParams = await paraSwap.buildTx(
-  srcToken,
-  destToken,
-  srcAmount,
-  destAmount,
-  priceRoute,
-  senderAddress,
-  referrer,
-  receiver,
-);
+const minParaSwap = constructPartialSDK({
+  network: 1,
+  fetcher,
+}, constructGetRate, constructGetAllowance)
 
-web3.eth.sendTransaction(
-  txParams,
-  async (err: Error, transactionHash: string) => {
-    if (err) {
-      return this.setState({ error: err.toString(), loading: false });
-    }
-    console.log('transactionHash', transactionHash);
-  },
-);
+const priceRoute = await minParaSwap.getRate(params)
+const allowance = await minParaSwap.getAllowance(userAddress, tokenAddress);
 ```
 
----
-
-### To run the example locally:
+## Legacy
+The ParaSwap class is exposed with some effort on backward compatibility with previous versions.
 
-Created an .env file with these 2 env variables:
 
-```bash
-PROVIDER_URL=YOUR_PROVIDRER_URL_OR_INFURA_URL
-NODE_ENV=production
-```
-
-run
+```typescript
+import { ParaSwap } from '@paraswap/sdk'
+import axios from 'axios'
+import Web3 from 'web3'
+
+const web3Provider = new Web3(window.ethereum)
+const account = '__user_address__'
+
+const paraswap = new ParaSwap(
+  1, 
+  undefined, 
+  web3Provider, 
+  undefined, 
+  account, 
+  axios
+)
 
-```bash
-yarn install paraswap
 ```
 
-For local developement you can run
-
-```bash
-yarn dev
-```
+By analogy to ```constructPartialSDK```, you can leverage a lightweight version of the sdk for fetching only
```

Comment:
```suggestion
Which corresponds to ```constructPartialSDK``` in the SDK's current version to leverage a lightweight version for fetching only.
```

Reply:
That's not exactly what I meant here. In the example above, we show how to use the ParaSwap class like before (with web3, axios). In this second example we show we can use the ParaSwap with 0 heavy dependancy which is sort of analogous to ```constructPartialSDK``` (with less bundle weight overhead though)